kernel/data-pipeline/dialect/mysql/src/main/java/org/apache/shardingsphere/data/pipeline/mysql/ingest/client/MySQLClient.java

```diff
@@ -298,14 +298,18 @@ public void exceptionCaught(final ChannelHandlerContext ctx, final Throwable cau
             log.error("MySQLBinlogEventHandler protocol resolution error, file name:{}, position:{}", fileName, position, cause);
             reconnect();
         }
-        
+    
         private void reconnect() {
             if (reconnectTimes.get() > 3) {
                 log.warn("exceeds the maximum number of retry times, last binlog event:{}", lastBinlogEvent);
                 running = false;
                 return;
             }
             int retryTimes = reconnectTimes.incrementAndGet();
+            if (null == lastBinlogEvent.getFileName()) {
+                log.warn("can't get file name from binlog event， last binlog event:{}", lastBinlogEvent);
+                return;
+            }
```

Comment:
How it occur?

Reply:
somethimes ,the lastBinlogEvent isn's set value after `connect()`, then if execute stop migration the `reconnect()` method will execute, will cause NPE at `subscribe(lastBinlogEvent.getFileName(), lastBinlogEvent.getPosition());`

Reply:
OK.
Looks `lastBinlogEvent.getPosition()` has similar issue, could we verify it too?